tests/main.js

```diff
@@ -651,6 +651,47 @@ test('subscribe to room and receive sent messages', t => {
   t.timeoutAfter(TEST_TIMEOUT)
 })
 
+test('send message with malformed attachment fails', t => {
+  fetchUser(t, 'alice')
+    .then(alice => alice.sendMessage({
+      roomId: bobsRoom.id,
+      text: 'should fail',
+      attachment: { some: 'rubbish' }
+    }))
+    .catch(err => {
+      t.true(toString(err).match(/attachment/), 'attachment error')
+      t.end()
+    })
+  t.timeoutAfter(TEST_TIMEOUT)
+})
+
+test(`send message with link attachment [sends one message to Bob's room]`, t => {
+  fetchUser(t, 'alice')
+    .then(alice => alice.sendMessage({
+      roomId: bobsRoom.id,
+      text: 'see attached',
+      attachment: { link: 'https://cataas.com/cat', type: 'image' }
+    }))
+    .then(() => t.end())
+    .catch(endWithErr(t))
+  t.timeoutAfter(TEST_TIMEOUT)
+})
```

Comment:
Are we missing a case for data attachments?

Reply:
done